scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt

```diff
@@ -134,11 +134,7 @@ class ClearlyDefinedStorage(
         try {
             runBlocking { readFromClearlyDefined(id, packageCoordinates(id, vcs, sourceArtifact)) }
         } catch (e: IllegalArgumentException) {
-            e.showStackTrace()
-
-            log.warn { "Could not obtain ClearlyDefined coordinates for package '${id.toCoordinates()}'." }
-
-            emptyResult(id)
+            handleException(id, e)
```

Comment:
So, if I'm not mistaken, this will cause the scanner to fail hard if the package identifier is not supported by ClearlyDefined, as the `IllegalArgumentException` is not catched / converted to a `ScanException` anywhere. If that's the case, I believe it's not what we want. If a package identifier is not supported by ClearlyDefined, it means it also cannot provide an curations for that package, and we should treat it the same way as a non-existing curation IMO.

Reply:
No, the exception is still catched, it's just handled the same way as the other exceptions in this class. The only change is that in this case now `Failure` is returned instead of `Success`, but that doesn't make the scanner fail.

Reply:
You are right, it's not unexpected, I'll remove that commit from the PR. It seems a bit risky to rely on the assumption that this is the only possibility for an `IllegalArgumentException` here, but I'm also generally not a fan of using exceptions for control flow. I think I'd prefer if `packageCoordinates()` would return null instead, but I also don't want to address this now.

Reply:
> Does this change has any effect on the behaviour of the scanner? If now every time this exception occurs an error message is logged, this is probably not desired, is it?

It now logs an error instead of a warning, but that's obsolete as I will revert that change.

Reply:
Done.